trade-finance-manager-ui/server/controllers/case/underwriting/amendments/amendment-lead-underwriter/index.js

```diff
@@ -0,0 +1,175 @@
+  }
+
+  // checks if user has ability to assign or change lead-underwriter to render change link or add button
+  const userCanEdit = canUserEditLeadUnderwriter(user);
+
+  return {
+    userCanEdit,
+    activePrimaryNavigation: 'manage work',
+    activeSubNavigation: 'underwriting',
+    deal: deal.dealSnapshot,
+    tfm: deal.tfm,
+    dealId: deal.dealSnapshot._id,
+    user,
+    currentLeadUnderWriter,
+    amendment,
+  };
+};
+
+/**
+ * @param {*} req
+ * @param {*} res
+ * gets deal and amendment by id
+ * checks if user can edit
+ * gets all underwriter/ managers and populates dropdown list
+ * renders template
+ */
+const getAssignAmendmentLeadUnderwriter = async (req, res) => {
+  const {
+    _id: dealId,
+    amendmentId,
+    facilityId,
+  } = req.params;
+  const deal = await api.getDeal(dealId);
+
+  const { data: amendment } = await api.getAmendmentById(facilityId, amendmentId);
+
+  // if no amendment or deal
+  if ((!amendment || !amendment.amendmentId) || !deal) {
+    return res.redirect('/not-found');
+  }
+
+  const { user } = req.session;
+
+  // checks if can edit
+  const userCanEdit = canUserEditLeadUnderwriter(user);
+
+  if (!userCanEdit) {
+    return res.redirect('/not-found');
+  }
+
+  let currentLeadUnderWriterUserId;
+
+  // if already set then shows name of assigned underwriter
+  if (amendment.leadUnderwriter) {
+    currentLeadUnderWriterUserId = amendment.leadUnderwriter.userId;
+  }
+
+  // gets all underwriter and managers
+  const allUnderwriterManagers = await api.getTeamMembers(CONSTANTS.TEAMS.UNDERWRITER_MANAGERS);
+  const allUnderwriters = await api.getTeamMembers(CONSTANTS.TEAMS.UNDERWRITERS);
+
+  const allTeamMembers = [
+    ...allUnderwriterManagers,
+    ...allUnderwriters,
+  ];
+
+  // sorts alphabetically
+  const alphabeticalTeamMembers = sortArrayOfObjectsAlphabetically(allTeamMembers, 'firstName');
+
+  return res.render('case/underwriting/amendments/amendment-lead-underwriter/amendment-assign-lead-underwriter.njk', {
+    activeSubNavigation: 'underwriting',
+    deal: deal.dealSnapshot,
+    tfm: deal.tfm,
+    dealId: deal.dealSnapshot._id,
+    user,
+    assignToSelectOptions: mapAssignToSelectOptions(currentLeadUnderWriterUserId, user, alphabeticalTeamMembers),
+    amendment,
+  });
+};
+
+/**
+ * @param {*} req
+ * @param {*} res
+ * checks if unassigned or assigned to user
+ * if unassigned then does not make getUser api call
+ * redirects back to underwriting page
+ */
+const postAssignAmendmentLeadUnderwriter = async (req, res) => {
+  const {
+    _id: dealId,
+    amendmentId,
+    facilityId,
+  } = req.params;
+  const deal = await api.getDeal(dealId);
+
+  const { data: amendment } = await api.getAmendmentById(facilityId, amendmentId);
+
+  if ((!amendment || !amendment.amendmentId) || !deal) {
+    return res.redirect('/not-found');
+  }
+
+  const { user } = req.session;
+
+  const userCanEdit = canUserEditLeadUnderwriter(user);
+
+  if (!userCanEdit) {
```

Comment:
same as above - the user should be able to see the page but should not be able to make changes

Reply:
side note, what happens if the deal is completed but you land on the page again (i.e. if you have saved the URL)?
you should check what I've done for the other files

Reply:
done